dclab/kde/base.py

```diff
@@ -358,18 +359,21 @@ def get_scatter(self, xax="area_um", yax="deform", positions=None,
         xs = self.apply_scale(x, xscale, xax)
         ys = self.apply_scale(y, yscale, yax)
 
-        if positions is None:
-            posx = None
-            posy = None
-        else:
-            posx = self.apply_scale(positions[0], xscale, xax)
-            posy = self.apply_scale(positions[1], yscale, yax)
-
-        kde_fct = methods[kde_type]
         if len(x):
-            density = kde_fct(events_x=xs, events_y=ys,
-                              xout=posx, yout=posy,
-                              **kde_kwargs)
+            xr, yr, density_grid = self.get_raster(xax=xax,
+                                                   yax=yax,
+                                                   kde_type=kde_type,
+                                                   kde_kwargs=kde_kwargs,
+                                                   xscale=xscale,
+                                                   yscale=yscale)
+            # 'scipy.interp2d' has been removed in SciPy 1.14.0
+            # https://scipy.github.io/devdocs/tutorial/interpolate/interp_transition_guide.html
+            interp_func = RGI((xr[:, 0], yr[0, :]),
+                              density_grid,
+                              method='linear',
+                              bounds_error=False,
+                              fill_value=0)
```

Comment:
Are we expecting out-of-bounds errors? If yes, the fill value should probably be `np.nan`, right?

Reply:
Yes, but assestions like `assert np.all(a[:20] == a[0])` and `assert np.all(a == b)` are failed because `np.nan == np.nan` is False. So, I used `fill_value=0` to avoid.

Reply:
Do the tests pass when you set bounds_error to True?